[16.0][IMP] account_invoice_ubl: add support to self-billing

jbaudoux opened the pull request:
Allow to generate self-billing invoice and credit note in UBL.

https://docs.peppol.eu/poacc/self-billing/3.0/bis-sb/

cc @lmignon @rousseldenis @sbidoul 

Also fixed in:
* account_invoice_ubl:
  * due date : only declare if there is an amount_residual

jbaudoux committed:
[IMP] account_invoice_ubl: add support to self-billing

jbaudoux committed:
[FIX] account_invoice_ubl: due date

The due date must always be there when there is a residual amount

lmignon commented:
IMO, this addon should depend on [account_invoice_supplier_self_invoice](https://github.com/OCA/account-invoicing/tree/16.0/account_invoice_supplier_self_invoice). Personally, I would have preferred the term `self_billing` to be used in the name of this existing module and for the fields defined in it instead of `self_invoice`, but despite this, it would seem logical to me to depend on this module.

jbaudoux committed:
account_invoice_ubl: taxes

- Fix non-VAT and exempt taxes
- Provide the tax description for non-VAT taxes converted to AllowanceCharge
- Do not provide VAT Party identifiers when the invoice is outside scope of VAT

jbaudoux committed:
[FIX] base_ubl: fix exempt taxes (AE, E, G, O)

Provide proper exemption reason.
Do not provide VAT Party identifiers when the invoice is outside scope of VAT. Add the company registry.

jbaudoux commented:
Includes https://github.com/OCA/edi/pull/1287
To rebase after that other PR is merged
